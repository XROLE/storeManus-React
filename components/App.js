import React, { Component } from 'react';
import {
  BrowserRouter as Router, Route, Switch, Redirect,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import Home from './home/Home';
import store from '../store';
import SignIn from './auth/SignIn';
import AdminDashboard from './admin/AdminDashboard';
import AttendantDashboard from './attendants/AttendantDashboard';
import AdminAddAttendant from './admin/AddAttendant';
import ProfileUpdate from './attendants/ProfileUpdate';

class App extends Component {
  constructor(props) {
    super();
  }

  render() {
    return (
      <Provider store={store}>
        <Router>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/login" component={SignIn} />
            <Route path="/adminDashboard" component={AdminDashboard} exact />
            <Route path="/admin/add/attendant" component={AdminAddAttendant} exact />
            <Route path="/attendantsDashboard" component={AttendantDashboard} exact />
            <Route path="/attendant/profile/update" component={ProfileUpdate} exact />
          </Switch>
        </Router>
      </Provider>
    );
  }
}

export default App;
