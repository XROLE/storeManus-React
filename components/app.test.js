import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App component', () => {
  it('Should render just the Navbar component', () => {
    const component = shallow(
      <App />,
    );
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
