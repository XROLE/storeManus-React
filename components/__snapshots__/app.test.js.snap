// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render just the Navbar component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/adminDashboard"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/add/attendant"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/attendantsDashboard"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/attendant/profile/update"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/product/add"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/products"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/products/available"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/products/finished"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/attendants"
      />
    </Switch>
  </BrowserRouter>
</Provider>
`;
